lib/internal/source_map.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+// See https://sourcemaps.info/spec.html for SourceMap V3 specification.
+
+const fs = require('fs');
+const debug = require('internal/util/debuglog').debuglog('source_map');
+const { dirname, resolve } = require('path');
+const {
+  normalizeReferrerURL,
+} = require('internal/modules/cjs/helpers');
+const { JSON, Object } = primordials;
+const sourceMapCache = new Map();
+const { URL } = require('url');
+
+function maybeCacheSourceMap(filename, content) {
+  try {
+    filename = normalizeReferrerURL(filename);
+  } catch (err) {
+    // This is most likely an [eval]-wrapper, which is currently not
+    // supported.
+    debug(err.message);
+    return;
+  }
+
+  const base = dirname((new URL(filename)).pathname);
+
+  if (!process.env.NODE_V8_COVERAGE) return;
+  const match = content.match(/\/[*/]#\s+sourceMappingURL=(?<sourceMappingURL>[^\s]+)/);
+  if (match) {
+    sourceMapCache.set(filename.replace('file://', ''), {
+      url: match.groups.sourceMappingURL,
+      data: dataFromUrl(base, match.groups.sourceMappingURL)
+    });
+  }
+}
+
+function dataFromUrl(base, sourceMappingURL) {
+  // Remove the scheme from file:// paths, so that they can be processed
+  // identically to paths without a scheme (both patterns are possible).
+  if (sourceMappingURL.match(/^file:\/\//)) {
+    sourceMappingURL = sourceMappingURL.replace(/^file:\/\//, '');
+  }
+  if (sourceMappingURL.match(/^data:/)) {
+    // TODO(bcoe): load source-map from base64 encoded data URL.
+  } else if (sourceMappingURL.match(/^[a-z0-9+-.]+:/)) {
+    // We currently don't support any schemes other than "data" and "file".
+    return null;
+  } else {
+    // If no scheme is present, we assume we are dealing with a file path:
+    const sourceMapFile = resolve(base, sourceMappingURL);
+    const content = fs.readFileSync(sourceMapFile, 'utf8');
+    const data = JSON.parse(content);
```

Comment:
I think we should handle the read error and parse error here?

Reply:
I would be tempted to just resolve `null`, and output a `debug` message if we fail to parse or read the source-map.